lib/ledger_sync/xero/account/operations/update.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+module LedgerSync
+  module Xero
+    class Account
+      module Operations
+        class Update < Xero::Operation::UpdatePost
+          class Contract < LedgerSync::Ledgers::Contract
+            params do
+              required(:external_id).filled(:string)
+              required(:ledger_id).filled(:string)
+              required(:Code).maybe(:string)
+              required(:Name).maybe(:string)
+              required(:BankAccountNumber).maybe(:string)
+              # required(:Description).maybe(:string)
+              required(:CurrencyCode).maybe(:string)
+              required(:Type).maybe(:string)
+              required(:Status).maybe(:string)
+              required(:BankAccountType).maybe(:string)
+              # required(:TaxType).maybe(:string)
+              # required(:Class).maybe(:string)
+            end
+
+            rule(:Status) do
```

Comment:
Is this is the right way to add validation (It doesn't fire) ? How do we test this condition, should it be in serialiser ?

Reply:
No, I wouldn't validate this. We need to validate that they aren't changing both at the same time, but the issue is that the API is the source of truth on the current state, not the resource. I would just let the API error for now. Don't worry too much about catching it.  We should just file this as an issue we know about and can address downstream.